TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/QueryHandlers/SetQtlsDateHandler.cs

```diff
@@ -0,0 +1,34 @@
+using Microsoft.PowerPlatform.Dataverse.Client;
+using Microsoft.Xrm.Sdk.Messages;
+using TeachingRecordSystem.Core.Dqt.Queries;
+
+namespace TeachingRecordSystem.Core.Dqt.QueryHandlers;
+
+public class SetQtlsDateHandler : ICrmQueryHandler<SetQtlsDateQuery, bool>
+{
+    public async Task<bool> Execute(SetQtlsDateQuery query, IOrganizationServiceAsync organizationService)
+    {
+        if (query.HasActiveSanctions == true)
+        {
+            await organizationService.CreateAsync(new CrmTask()
+            {
+                RegardingObjectId = query.ContactId.ToEntityReference(Contact.EntityLogicalName),
+                Category = "QTLS date set/removed for record with an active alert",
+                Description = !query.QtlsDate.HasValue ? "QTLSDate removed for a record with active alert" : $"QTLSDate {query.QtlsDate} set for a record with active alert",
+                Subject = "Notification for SET QTLS data collections team",
+                ScheduledEnd = query.TaskScheduleEnd
+            });
+        }
+
+        await organizationService.ExecuteAsync(new UpdateRequest()
```

Comment:
These two calls need to be inside a transaction request (see `CreateContactHandler`)